mlprimitives/candidates/dsp.py

```diff
@@ -0,0 +1,211 @@
+        """Inits SpectralMask Class with default values for method, gain, window_length, and
+             beta
+
+        Args:
+            method: Method used to calculate the Spectral Mask: 'std_dev' uses the standard
+              deviation of each frequency component and 'min_max' uses the minimum and maximum
+              values
+            gain: Multiplication factor used in the comparison between the spectral mask defined
+              using the training data and the Fourier transform of the telemetry data vector
+            window_length: Length of the sliding window in number of samples
+            beta: Beta value for Kaiser window design
+
+        """
+
+        self.gain = gain
+        self.beta = beta
+        self.window_length = window_length
+        self.window = None
+        self.method = method
+        self.window_length = next_power_of_2(window_length)
+
+    def window_design(self, window_length=None, beta=None):
+        """Kaiser window design
+
+        Args:
+            window_length: Length of the window in number of samples
+            beta: Beta value for Kaiser window design
+
+        Returns:
+            window: Window designed using the beta and length provided as inputs
+
+        """
+
+        if window_length is None:
+            window_length = self.window_length
+        else:
+            self.window_length = window_length
+
+        if beta is None:
+            beta = self.beta
+        else:
+            self.beta = beta
+
+        self.window = np.kaiser(window_length, beta)
+
+        return self.window
+
+    def fit(self, X, method=None, gain=None, window_length=None, beta=None):
+        """Defines a spectral mask based on training data
+
+        Args:
+            X: Training data
+            method: Method used to calculate the Spectral Mask: 'std_dev' uses the standard
+              deviation of each frequency component and 'min_max' uses the minimum and maximum
+              values
+            gain: Multiplication factor used in the comparison between the spectral mask defined
+              using the training data and the Fourier transform of the telemetry data vector
+            window_length: Length of the sliding window in number of samples
+            beta: Beta value for Kaiser window design
+
+        """
+
+        training_signal = X
+
+        if gain is None:
+            gain = self.gain
+        else:
+            self.gain = gain
+
+        if window_length is None:
+            window_length = self.window_length
+        else:
+            self.window_length = window_length
+
+        if beta is None:
+            beta = self.beta
+        else:
+            self.beta = beta
+
+        if method is None:
+            method = self.method
+        else:
+            self.method = method
+
+        self.window_design(window_length, beta)
+
+        if method == 'std_dev':
+            self.fit_freq_std_dev(training_signal)
+        else:

```

Comment:
I think it would be better to have a strict check here, and raise an exception if the value is not one of the expected ones:

```
if method == 'std_dev':
    self.fit_freq_std_dev(training_signal)
elif method == 'min_max':
    self.fit_freq_min_max(training_signal)
else:
    raise ValueError('Unknown method: {}'.format(method))
```

Reply:
will do, thanks.